pkgs/development/python-modules/rsa/default.nix

```diff
@@ -1,36 +1,47 @@
 { lib
 , buildPythonPackage
-, fetchPypi
-, unittest2
+, fetchFromGitHub
+, mypy
+, poetry-core
 , pyasn1
-, mock
-, isPy3k
+, pytest-cov
+, pytestCheckHook
 , pythonOlder
 }:
 
 buildPythonPackage rec {
   pname = "rsa";
   version = "4.9";
+  format = "pyproject";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-44RkpJxshdfxNRsBJmYUh6fgoUpQ8WdexQ6zTU8g7yE=";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "sybrenstuvel";
+    repo = "python-rsa";
+    rev = "version-${version}";
+    hash = "sha256-PwaRe+ICy0UoguXSMSh3PFl5R+YAhJwNdNN9isadlJY=";
   };
 
-  checkInputs = [ unittest2 mock ];
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
   propagatedBuildInputs = [ pyasn1 ];
 
-  preConfigure = lib.optionalString (isPy3k && pythonOlder "3.7") ''
+  checkInputs = [
+    mypy
+    pytest-cov
```

Comment:
PLease patch out the coverage parts. This doesn't add value for us.

Reply:
Done, thanks for the feedback.